query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -321,7 +322,8 @@ private SearchIntegrator getSearchFactory(String cacheName, IndexingConfiguratio
       //defend against multiple initialization:
       if (searchFactory == null) {
          ClassLoader aggregatedClassLoader = makeAggregatedClassLoader(cr.getGlobalComponentRegistry().getGlobalConfiguration().classLoader());
-         Collection<ProgrammaticSearchMappingProvider> programmaticSearchMappingProviders = ServiceFinder.load(ProgrammaticSearchMappingProvider.class, aggregatedClassLoader);
+         Collection<ProgrammaticSearchMappingProvider> programmaticSearchMappingProviders = new HashSet<>(ServiceFinder.load(ProgrammaticSearchMappingProvider.class, aggregatedClassLoader));
```

Comment:
Why add it manually and not use the regular mechanism for DefaultSearchMappingProvider ?

Reply:
The final effect is the same, this makes it simpler and it works across all kinds of classloaders (osgi, wildfly, uberjar, etc)